mp/src/game/client/momentum/ui/SettingsPanel/MomentumSettingsDialog.cpp

```diff
@@ -245,4 +247,6 @@ CON_COMMAND_F(mom_settings_show, "Shows the settings panel.\n", FCVAR_CLIENTCMD_
     }
 
     g_pSettingsDialog->DoModal();
+
+    engine->ClientCmd_Unrestricted("gameui_preventescapetoshow\n");
```

Comment:
As per your description, if this wasn't the fix to it this commit can be dropped, yeah. The chat panel does it so that escape closes chat and doesn't open the menu. The settings menu being opened already means the menu is open, so this command was just a red herring.